server/app/mutations/host_nodes/health_check.rb

```diff
@@ -0,0 +1,36 @@
+module HostNodes
+  class HealthCheck < Mutations::Command
+
+    include Common
+
+    required do
+      model :host_node
+    end
+
+    def validate
+      unless self.host_node.connected?
+        add_error(:connection, :disconnected, self.host_node.websocket_error)
+      end
+    end
+
+    def rpc_client
+      @rpc_client ||= self.host_node.rpc_client(10)
+    end
+
+    def query_etcd_health
+      begin
+        return rpc_client.request("/etcd/health")
+      rescue RpcClient::TimeoutError => error
+        add_error(:etcd_health, :timeout, error.message)
+
+        return {health: nil, error: nil}
```

Comment:
This ends up getting ignored after the `add_error`, and the `{error: ...}` here is misleading.